### PR TITLE
refactor: move theme plugin into experience package

### PR DESCRIPTION
**Pull request by supproduction**
moves theme\resource plugins into experience package
adds  utilities for media
updated theme interfaces
unifies sizes interface

closes [HRZ-2787](https://spryker.atlassian.net/browse/HRZ-2787)

[HRZ-2787]: https://spryker.atlassian.net/browse/HRZ-2787?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/base/ui/form/select/src/stories/select-demo.stories.ts [hunk 1]
```diff
@@ -125,7 +124,7 @@ SelectDemo.parameters = {
 
 SelectDemo.args = {
   maxHeight: '300px',
-  clearIcon: IconTypes.Remove,
+  clearIcon: 'remove',
```
Comment (tobi-or-not-tobi) [inline libs/base/ui/form/select/src/stories/select-demo.stories.ts]: why not keep the enum value?

File: libs/template/cli/bin/cli.ts [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+import { CliPlugin, cliProviders } from '@spryker-oryx/cli';
+import {
+  appBuilder,
+  coreFeature,
+  FeatureOptions,
+  InjectionPlugin,
+} from '@spryker-oryx/core';
 
-cliApp.withOptions({ cli: { args: process.argv.slice(2) } }).create();
+// TODO: fix it
```
Comment (tobi-or-not-tobi) [inline libs/template/cli/bin/cli.ts]: what is there to fix?

File: libs/template/themes/breakpoints/src/default.breakpoints.ts [hunk 3]
```diff
@@ -1,7 +1,6 @@
-import { ThemeBreakpoints } from '@spryker-oryx/core';
-import { Size } from '@spryker-oryx/utilities';
+import { Breakpoints, Size } from '@spryker-oryx/utilities';
 
-export const defaultBreakpoints: ThemeBreakpoints = {
+export const defaultBreakpoints: Breakpoints = {
   [Size.Sm]: {
     min: 0,
```
Comment (tobi-or-not-tobi) [inline libs/template/themes/breakpoints/src/default.breakpoints.ts:5]: of topic, but I assume this one sh/could go

**Review by tobi-or-not-tobi — DISMISSED** [review]

**Review by tobi-or-not-tobi — DISMISSED** [review]

**Review by tobi-or-not-tobi — DISMISSED** [review]

**Review by tobi-or-not-tobi — APPROVED** [review]

File: libs/platform/experience/src/plugins/theme/theme.ts [hunk 4]
```diff
@@ -43,14 +50,20 @@ export class ThemePlugin extends ThemeTokens implements AppPlugin {
...
   }
 
   getIcon(icon: string): string | Promise<string> {
     return resolveLazyLoadable(this.icons[icon]);
   }
 
+  beforeApply(app: App): void {
+    const components = app.findPlugin(ComponentsPlugin);
+    components?.setResolver({ themes: this.resolve.bind(this) });
+    components?.extendComponent(this.applyThemes.bind(this));
```
Comment (gund) [inline libs/platform/experience/src/plugins/theme/theme.ts:64]: Use `requirePlugin` to enforce `ComponentsPlugin` and remove optional chaining as ThemePlugin cannot working without it anyway:
```ts
    const components = app.requirePlugin(ComponentsPlugin);
    components.setResolver({ themes: this.resolve.bind(this) });
    components.extendComponent(this.applyThemes.bind(this));
```

File: libs/platform/core/src/orchestration/app/app.model.ts [hunk 5]
```diff
@@ -11,18 +8,11 @@ declare global {
...
 }
 
-export interface ModularAppBuilderOptions {
-  injector?: Omit<InjectorOptions, 'providers'>;
-  components?: Partial<ComponentsOptions>;
-}
-
+// TODO: move out to the experience package
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface AppFeature {
```
Comment (gund) [inline libs/platform/core/src/orchestration/app/app.model.ts:13]: Why should `AppFeature` be moved to experience? It's a core part of the framework.

File: libs/template/application/src/app/app-builder.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import { AppBuilderWithModules } from './app.model';
+import { ModularAppBuilder } from './modular-app-builder';
+
+export function appBuilder(): AppBuilderWithModules {
+  return new ModularAppBuilder();
```
Comment (gund) [inline libs/template/application/src/app/app-builder.ts:5]: I'm not sure why do we need 2 separate appBuilders in core and application packages?
We had only one before, what's the reason to have 2 now?

**Pull request merged**